Gradle Language Server doesn't work with Language Support for Java extension

Extension Name: vscode-gradle
Extension Version: 0.17.2
OS Version: Win 11
VSCode version: 1.103.2 (Kiro 0.8.140)
Java Version: 21.0.9

**Describe the bug**
Any attempt to modify build.gradle file in java project with Language Support for Java extension active produces following exception:

```
java.lang.SecurityException: class "org.eclipse.lsp4j.util.Ranges"'s signer information does not match signer information of other classes in the same package
	at java.base/java.lang.ClassLoader.checkCerts(Unknown Source)
	at java.base/java.lang.ClassLoader.preDefineClass(Unknown Source)
	at java.base/java.lang.ClassLoader.defineClass(Unknown Source)
	at java.base/java.security.SecureClassLoader.defineClass(Unknown Source)
	at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(Unknown Source)
	at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(Unknown Source)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(Unknown Source)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(Unknown Source)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(Unknown Source)
	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
	at com.microsoft.gradle.GradleServices.completion(GradleServices.java:275)
	... 14 more
```

This seems to be caused by #1733 as gradle extension is using ridiculously outdated lsp4j (0.19.0 from Nov of 2022) while Language Support for Java runs current 0.24.0. I suspect a certificate change somewhere in between.

I haven't found any version of Language Support for Java extension that would work with this. 0.17.1 version of the Gradle extension works with limited autocomplete. It will not autocomplete implementation but will autocomplete group, name and version in following code:

```
dependencies {
    implementation (group: "org.springframework.boot", name: "spring-boot-starter-web", version: "4.0.2")
}
```

**To Reproduce**
Create Java Gradle project with Language Support for Java extension and try to modify any build.gradle file.

**Expected behavior**
Attempt to autocomplete shouldn't produce any exceptions.

**Screenshots**


**Output from "Gradle for Java"**
I masked some of the filesystem paths by *** but that shouldn't impact anything.
```
[info] [gradle-server] Gradle Server started, listening on 56464
[info] Gradle client connected to server
[info] Java Home: C:\Program Files\Amazon Corretto\jdk21.0.9_10
[info] JVM Args: -XX:MaxMetaspaceSize=384m,-XX:+HeapDumpOnOutOfMemoryError,-Xms256m,-Xmx512m,-Dfile.encoding=UTF-8,-Duser.country=US,-Duser.language=en,-Duser.variant
[info] Gradle User Home: C:\Users\***\.gradle
[info] Gradle Version: 9.3.0
[info] > Configure project :
Project version is 1.0.0-SNAPSHOT

[Incubating] Problems report is available at: file:///C:/***/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.3.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

CONFIGURE SUCCESSFUL in 5s
[info] Found 245 tasks
[info] [gradle-server] Jan 24, 2026 1:34:50 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint fallbackResponseError
SEVERE: Internal error: java.lang.reflect.InvocationTargetException
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:67)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.request(GenericEndpoint.java:120)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:261)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:190)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
	... 11 more
Caused by: java.lang.SecurityException: class "org.eclipse.lsp4j.util.Ranges"'s signer information does not match signer information of other classes in the same package
	at java.base/java.lang.ClassLoader.checkCerts(Unknown Source)
	at java.base/java.lang.ClassLoader.preDefineClass(Unknown Source)
	at java.base/java.lang.ClassLoader.defineClass(Unknown Source)
	at java.base/java.security.SecureClassLoader.defineClass(Unknown Source)
	at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(Unknown Source)
	at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(Unknown Source)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(Unknown Source)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(Unknown Source)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(Unknown Source)
	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
	at com.microsoft.gradle.GradleServices.completion(GradleServices.java:275)
	... 14 more
```

**Does the bug still exist if you disable all other extensions?**
No

**Additional context**

--- a/cgmanifest.json
+++ b/cgmanifest.json
@@ -7,7 +7,7 @@
         "Maven": {
           "GroupId": "org.eclipse.lsp4j",
           "ArtifactId": "org.eclipse.lsp4j",
-          "Version": "0.12.0"
+          "Version": "0.24.0"
         }
       },
       "DevelopmentDependency": false
@@ -18,7 +18,7 @@
         "Maven": {
           "GroupId": "org.eclipse.lsp4j",
           "ArtifactId": "org.eclipse.lsp4j.jsonrpc",
-          "Version": "0.12.0"
+          "Version": "0.24.0"
         }
       },
       "DevelopmentDependency": false

--- a/gradle-language-server/build.gradle
+++ b/gradle-language-server/build.gradle
@@ -21,8 +21,9 @@ application {
 }
 
 dependencies {
-  implementation "org.eclipse.lsp4j:org.eclipse.lsp4j:0.19.0"
-  implementation "org.eclipse.lsp4j:org.eclipse.lsp4j.jsonrpc:0.19.0"
+  implementation "org.eclipse.lsp4j:org.eclipse.lsp4j:0.24.0"
+  implementation "org.eclipse.lsp4j:org.eclipse.lsp4j.jsonrpc:0.24.0"
+  implementation "com.google.guava:guava:32.1.3-jre"
   implementation "org.codehaus.groovy:groovy-eclipse-batch:4.0.16-03"
   implementation "com.google.code.gson:gson:2.9.1"
   implementation "org.apache.bcel:bcel:6.6.1"

--- a/gradle-server/build.gradle
+++ b/gradle-server/build.gradle
@@ -29,8 +29,8 @@ dependencies {
   implementation "io.grpc:grpc-netty:${grpcVersion}"
   implementation 'io.github.g00fy2:versioncompare:1.4.1'
   implementation project(':gradle-language-server')
-  implementation "org.eclipse.lsp4j:org.eclipse.lsp4j:0.19.0"
-  implementation "org.eclipse.lsp4j:org.eclipse.lsp4j.jsonrpc:0.19.0"
+  implementation "org.eclipse.lsp4j:org.eclipse.lsp4j:0.24.0"
+  implementation "org.eclipse.lsp4j:org.eclipse.lsp4j.jsonrpc:0.24.0"
   runtimeOnly 'org.slf4j:slf4j-simple:2.0.0-alpha6'
   testImplementation "io.grpc:grpc-testing:${grpcVersion}"
   testImplementation 'junit:junit:4.13.1'

